presto-hive-common/src/main/java/com/facebook/presto/hive/HdfsContext.java

```diff
@@ -24,6 +24,7 @@
 
 public class HdfsContext
 {
+    public static final String METASTORE_HEADERS = "metastore_headers";
```

Comment:
we dont need this right.

Reply:
We need this as this is in hive-commons. And in hive-commons there are places that requires creation of MetastoreContext and HiveSessionProperties field is not accessible. 
I am confused actually why HiveSessionProperties is not at the hive-commons?

Reply:
Do we really need this method here. Metastore headers in HdfsContext seems improper.
I see getMetastoreHeaders() in HdfsContext is being called from SemiTransactionalHiveMetastore alone. 

Why dont we do `action.getContext().getSession()` .
`MetastoreContext metastoreContext = new MetastoreContext(action.getContext().getIdentity(), action.getContext().getQueryId().orElse(""), action.getContext().getClientInfo(), action.getContext().getSource(), action.getContext().getSession().map(session -> HiveSessionProperties:getMetastoreHeaders));`

Reply:
Ah this might need to cyclic dependency between `presto-hive-metastore` and `presto-hive`. 

So may be add `getMetastoreHeaders()` in `MetastoreUtil.java` similar to what we have in `HiveSessionProperties.java`. Then we can probably do `action.getContext().getSession().map(session -> MetastoreUtil:getMetastoreHeaders) `

This is not a clean solution but seems inevitable.

Reply:
👍